### PR TITLE
fix(rpc): remove duplicate fields from rpc state

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

This PR removes `chain_store` and `beacon` fields from `RPCState` that are accessible from `state_manager`.

Changes introduced in this pull request:

- removes `chain_store` and `beacon` fields from `RPCState`
- add methods to `RPCState` for accessing `beacon`, `chain_store`, `chain_config` and `chain_index`
- publicly expose `StateManager::chain_rand` to avoid manually call `ChainRand::new` in RPC implementations
- cleanup and simplify RPC implementations code

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/state_manager/mod.rs [hunk 1]
```diff
@@ -1167,7 +1176,7 @@ where
 
     pub async fn miner_get_base_info(
         self: &Arc<Self>,
-        beacon_schedule: Arc<BeaconSchedule>,
+        beacon_schedule: &Arc<BeaconSchedule>,
```
Comment (lemmih) [inline src/state_manager/mod.rs:1179]:
```suggestion
        beacon_schedule: &BeaconSchedule,
```

**Review by lemmih — APPROVED** [review]
I am not super happy about the references to Arcs.

**Comment by hanabi1224** [comment]
> I am not super happy about the references to Arcs.

@lemmih I will see what I can do with a subsequent PR, to not further bloat this one.

**Review by aatifsyed — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Review by aatifsyed — COMMENTED** [review]
so much cleaner!

**Pull request merged**